---
title: "Exploratory analysis of a GEO dataset"
author: "NÃºria Garcia Deltiempo"
date: "15/5/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction:
The goal of this practical is to learn how to work with Markdown and Bioconductor and understand the steps that we follow to analyze our data. 

# Summary:

This study titled "Pancreatic cancer biomarkers in mouse saliva" is about to trying to detect potential salivary biomarkers in mice with a pancreatic tumor.
Global gene expression profiling has shown great promise in high-throughput biomarker discovery for early disease detection in body fluids such as saliva, which is accessible, cost-effective, and non-invasive. However, this goal has not been fully realized because saliva, like many clinical samples, contains partially fragmented and degraded RNAs that are difficult to amplify and detect with prevailing technologies. Here, using nanogram scale salivary RNA as a proof-of-principle example, we describe our progress with a novel poly-A tail independent mRNA amplification strategy combined with the Affymetrix GeneChip Exon arrays. We defined a Salivary Exon Core Transcriptome (SECT) with highly similar expression profiles in healthy individuals verified by quantitative PCR. Informatics analysis of SECT provided important mechanistic insight to their potential origin and function. Finally we demonstrated the diagnostic potential of true exon level expression profiling approach with salivary exon biomarkers that accurately discriminated gender in healthy individuals.
Recent studies have demonstrated that discriminatory salivary biomarkers can be readily detected upon the development of systemic diseases such as pancreatic cancer, breast cancer, lung cancer and ovarian cancer. However, the utility of salivary biomarkers for the detection of systemic diseases has been undermined due to the absence of biological and mechanistic rationale why distal diseases from the oral cavity would lead to the development of discriminatory biomarkers in saliva. Here, we examine the hypothesis that pancreatic tumor-derived exosomes are mechanistically involved in the development of pancreatic cancer-discriminatory salivary transcriptomic biomarkers. We first developed a pancreatic cancer mouse model that yielded discriminatory salivary biomarkers by implanting the mouse pancreatic cancer cell line Pan02 into the pancreas of the syngeneic host C57BL/6. The role of pancreatic cancer-derived exosomes in the development of discriminatory salivary biomarkers was then tested by engineered a Pan02 cell line that is suppressed for exosome biogenesis, implanted into the C56BL/6 mouse and examine if the discriminatory salivary biomarker profile was ablated or disrupted. Suppression of exosome biogenesis results in the ablation of discriminatory salivary biomarker development. This study supports that tumor-derived exosomes provide a mechanism in the development of discriminatory biomarkers in saliva and distal systemic diseases.

## Contributors:
Lau CS, Spielmann N, Lin YL, Eibl G, Elashoff D, Fang W, Moro A, Grogan T, Chiang S, Chia D, Wong DT

# Data description:

Here we are introducing the data that we are goig to work with.
```{r readTarget}
getwd()
target <- read.csv("~/Escritorio/BDBI/Second year/tercer trimestre/OMICS/Exercise 1 NOU/data/targets2.txt", sep= "")
expresion <- as.matrix(read.delim("/home/nuria/Escritorio/BDBI/Second year/tercer trimestre/OMICS/Exercise 1 NOU/data/expression2.txt", row.names = 1))
```


```{r}
dim(expresion)
summary(expresion)
boxplot(expresion)
```

Here we are installing the necessary package to handle the data that we have.

```{r}
require(Biobase)
```


```{r}
myEset <- ExpressionSet(expresion)
class(myEset)
show(myEset)

```

Here we are analyzing the covariates, to see how our data act.
```{r}
columnDesc <-  data.frame(labelDescription= c("SampleNames", "ShortName","Group"))
myAnnotDF <- new("AnnotatedDataFrame", data=target, varMetadata= columnDesc)
show(myAnnotDF)
```

```{r}
phenoData(myEset) <- myAnnotDF
```

```{r}
rownames(pData(myAnnotDF))<-pData(myAnnotDF)$SampleName
myEset <- ExpressionSet(assayData=expresion, phenoData=myAnnotDF)
show(myEset)
```

```{r}
myEset <- ExpressionSet(assayData=expresion, 
                        phenoData=myAnnotDF, 
                        featureNames =myGenes)
```

Here is where we start storing the data.
```{r}
myDesc <- new("MIAME", title="Pancreatic cancer biomarkers in mouse saliva")
print(myDesc)
```

```{r}
myEset <- ExpressionSet(assayData=expresion, 
                        phenoData=myAnnotDF,
                        fetureNames =myGenes,
                        experimentData = myDesc)
```

```{r}
dim(exprs(myEset))
class(phenoData(myEset))
class(pData(phenoData(myEset)))
head(pData(phenoData(myEset)))
head(pData(myEset))
```

Here we apply this functions to make data manipulation consistent.
```{r}
smallEset <- myEset[1:15,c(1:3,6:8)]
dim(exprs(smallEset))
dim(pData(smallEset))
head(pData(smallEset))
all(colnames(exprs(smallEset))==rownames(pData(smallEset)))
```

Here we are creating the dataset for the young patients, smaller than 30 years.
```{r}
youngEset <- myEset[,pData(myEset)$age<30]
dim(exprs(youngEset))
head(pData(youngEset))
```

```{r}
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE57820")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]


```

















